bin/check-dependency-version-consistency.ts

```diff
@@ -33,37 +33,44 @@ function collect(value: string, previous: string[]) {
 
 // Setup CLI.
 const program = new Command();
-program
-  .version(getCurrentPackageVersion())
-  .argument('<path>', 'path to workspace root')
-  .option(
-    '--fix',
-    'Whether to autofix inconsistencies (using highest version present)',
-    false
-  )
-  .option(
-    '--ignore-dep <dependency>',
-    'Dependency to ignore (option can be repeated)',
-    collect,
-    []
-  )
-  .action(function (path, options: { ignoreDep: string[]; fix: boolean }) {
-    // Calculate.
-    const dependencyVersions = calculateVersionsForEachDependency(path);
-    let mismatchingVersions = filterOutIgnoredDependencies(
-      calculateMismatchingVersions(dependencyVersions),
-      options.ignoreDep
-    );
 
-    if (options.fix) {
-      mismatchingVersions = fixMismatchingVersions(path, mismatchingVersions);
-    }
+try {
```

Comment:
We don't need to put all this code inside the `try`. Let's only put the `try` around this code as this is the primary logic that exceptions can be thrown from.

```js
const dependencyVersions = calculateVersionsForEachDependency(path);
let mismatchingVersions = filterOutIgnoredDependencies(
    calculateMismatchingVersions(dependencyVersions),
    options.ignoreDep
);

if (options.fix) {
    mismatchingVersions = fixMismatchingVersions(path, mismatchingVersions);
}
```

Reply:
@bmish how should we handle the error from `getCurrentPackageVersion()`?

Reply:
Good question. That shouldn't ever really happen. But to be safe, let's move to a new approach to ensure we capture all exceptions.

Move all CLI code into a new function called `run`. It will be much nicer to surround it with a try catch then.

```js
function run() {
   // ...
}

try {
  run();
} catch (e) {
  // ...
}
```